Adjust doctest for describe(G::GAPGroup)

fingolfin opened the pull request:
The output of `describe` is in general not unique and depends on the
state of a random number generator. The previous doctest used an
example that was affected by this, which we remove.

Also add an example involving a free group, just to show this other
flavor of the function.

fingolfin commented on src/Groups/GAPGroups.jl:
```diff
@@ -887,13 +887,13 @@ julia> describe(g)
 
 julia> describe(sylow_subgroup(g,2)[1])
-"C2 x ((((C2 x C2 x C2 x C2) : C2) : C2) : C2)"
+"(C2 x D8) : D8"
```
@ThomasBreuer CI tests fail for this and I have no idea why; do you? I've run `unique([describe(sylow_subgroup(symmetric_group(8),2)[1]) for i in 1:100])` a few times now and I've not yet once seen a different output. Yet in CI we get `(((C2 x C2 x C2 x C2) : C2) : C2) : C2`. 

I wonder if there is a global `ValueOption` we accidentally set somewhere? I briefly looked through the options used in the `StructureDescription` code but none of them explains this output. Nor why I can't reproduce it locally.

benlorenz replied:
I do get this locally:
```
julia> unique([describe(sylow_subgroup(symmetric_group(8),2)[1]) for i in 1:100])
1-element Vector{String}:
 "(((C2 x C2 x C2 x C2) : C2) : C2) : C2"
```
(on two different linux machines with julia 1.6.4 and 1.7.1)

fingolfin replied:
Ok, good to know (and weird)! I'll try it on different systems then. 

Aha perhaps it depends on which packages are installed and compiled. That gives me some ideas

fingolfin replied:
It is caused by the GAP package `crisp`. I have not yet investigate how or why.

ThomasBreuer replied:
The output of `SemidirectDecompositionsOfFiniteGroup( G )` seems to depend on whether GAP is running with or without packages, in the sense that the ordering of entries in the returned list changes.
In our situation, the function is called as `SemidirectDecompositionsOfFiniteGroup( G, "str" )` which apparently returns the first nontrivial decomposition as a semidirect product from the list of possibilities.
One could try to choose a "good" decomposition, but the problem itself would remain.

Perhaps the caveats in the documentation of `StructureDescription` should be extended; currently it is stated that different representations of the same group can lead to different description strings, but in fact a group with the same set of generators can lead to different description strings in different GAP sessions.

(By the way, calling `StructureDescription( s : recompute )` yields the variant `"(((C2 x D8) : C2) : C2) : C2"` if packages are loaded.)

ThomasBreuer replied:
@fingolfin My aim of the original doctest (enter `describe(sylow_subgroup(symmetric_group(10), 2)[1])`, get `"C2 x ((((C2 x C2 x C2 x C2) : C2) : C2) : C2)"`) was to give an example of the statement that "the results are still often not very helpful".

Due to the fact that the result also depends on several conditions, and that the same holds also in easier situations, this is apparently not a good idea. Perhaps we should simply omit this (intentionally) bad example.
(The same would hold also for the GAP test suite, where we want to run the tests in various situations, with and without packages.)

fingolfin committed:
Adjust doctest for `describe(G::GAPGroup)`

The output of `describe` is in general not unique and depends on the
state of a random number generator. The previous doctest used an
example that was affected by this, which we remove.

Also add an example involving a free group, just to show this other
flavor of the function.
